### DISCUSSION TITLE
Does `ClientWebSocketResponse.receive()` and `ClientWebSocketResponse.close()` ever raise an exception?

### DISCUSSION THREAD

**Discussion by RainerZufall187** [aio-libs/aiohttp#7337] [Q&A] [answered]
^^^^^^^^^^^^

**Comment by Dreamsorcerer** [comment, marked as answer]
Well, yes, there's a RuntimeError if you misuse it:
https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client_ws.py#L238

A TimeoutError if you use the timeout:
https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client_ws.py#L260

Looks like other exceptions are caught and turned into WSMessages though.

`close()` looks pretty safe, I wouldn't expect an exception to occur there.

**Comment by RainerZufall187** [comment]
ok so there is no need for try and except
